regtests/bin/matrix.base

```diff
@@ -695,19 +695,33 @@
-      echo "$rtst           -w work_PR3_UQ  -o netcdf   $ww3 ww3_tp2.8" >> matrix.body
-      echo "$rtst -s PR3_UQ -w work_PR3_UQ              $ww3 ww3_tp2.13" >> matrix.body
-      echo "$rtst -o netcdf -w work_PR3_UQ              $ww3 ww3_tp2.15" >> matrix.body
-      echo "$rtst -o netcdf -w work_5km -g 5km          $ww3 ww3_tp2.15" >> matrix.body
-      echo "$rtst -s TIDE   -w work_TIDE -o netcdf      $ww3 ww3_tp2.18" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ                 $ww3 ww3_tp2.1" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ                 $ww3 ww3_tp2.2" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ                 $ww3 ww3_tp2.3" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ                 $ww3 ww3_tp2.4" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ_curv -g curv    $ww3 ww3_tp2.4" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ                 $ww3 ww3_tp2.5" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ_a    -g a       $ww3 ww3_tp2.9" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ_b    -g b       $ww3 ww3_tp2.9" >> matrix.body
+      echo "$rtst -o netcdf              -w work_PR3_UQ                 $ww3 ww3_tp2.8" >> matrix.body
+      echo "$rtst -s PR3_UQ              -w work_PR3_UQ                 $ww3 ww3_tp2.13" >> matrix.body
+      echo "$rtst -o netcdf              -w work_PR3_UQ                 $ww3 ww3_tp2.15" >> matrix.body
+      echo "$rtst -o netcdf              -w work_5km -g 5km             $ww3 ww3_tp2.15" >> matrix.body
+      echo "$rtst -o netcdf -i input_rho -w work_PR3_UQ_RHO -r ww3_grid $ww3 ww3_tp2.15" >> matrix.body
+      echo "$rtst -o netcdf -i input_rho -w work_PR3_UQ_RHO -r ww3_strt $ww3 ww3_tp2.15" >> matrix.body
+      echo "cp ww3_tp2.15/input_rho/ww3_prnc_RHO.inp ww3_tp2.15/input_rho/ww3_prnc.inp" >> matrix.body
+      echo "$rtst -o netcdf -i input_rho -w work_PR3_UQ_RHO -r ww3_prnc $ww3 ww3_tp2.15" >> matrix.body
+      echo "mv ww3_tp2.15/work_PR3_UQ_RHO/ww3_prnc.out ww3_tp2.15/work_PR3_UQ_RHO/ww3_prnc_RHO.inp" >> matrix.body
+      echo "cp ww3_tp2.15/input_rho/ww3_prnc_TAU.inp ww3_tp2.15/input_rho/ww3_prnc.inp" >> matrix.body
+      echo "$rtst -o netcdf -i input_rho -w work_PR3_UQ_RHO -r ww3_prnc $ww3 ww3_tp2.15" >> matrix.body
+      echo "mv ww3_tp2.15/work_PR3_UQ_RHO/ww3_prnc.out ww3_tp2.15/work_PR3_UQ_RHO/ww3_prnc_TAU.inp" >> matrix.body
+      echo "cp ww3_tp2.15/input_rho/ww3_prnc_WND.inp ww3_tp2.15/input_rho/ww3_prnc.inp" >> matrix.body
+      echo "$rtst -o netcdf -i input_rho -w work_PR3_UQ_RHO -r ww3_prnc $ww3 ww3_tp2.15" >> matrix.body
+      echo "mv ww3_tp2.15/work_PR3_UQ_RHO/ww3_prnc.out ww3_tp2.15/work_PR3_UQ_RHO/ww3_prnc_WND.inp" >> matrix.body
+      echo "$rtst -o netcdf -i input_rho -w work_PR3_UQ_RHO -r ww3_shel $ww3 ww3_tp2.15" >> matrix.body
```

Comment:
I  got a segfault when running this regression test on our machines.  Also, is the reason for the multiple lines here and the cp statements because there is not logic in the run_test to run multiple ww3_prnc?   I also got segfaults running the ww3_ts3 tests.

Reply:
If these test are working for you, let me know and I'll start adding more debugging statements to help determine what is going on our machines; but I wanted to check with you first.

Reply:
I also have some errors. I will fix them and test the changes again, and come back to you when everything is working. I probably just missed something when changing the code for the conflicting new output fields.

Reply:
@ukmo-juan-castillo Thanks! Just let me know when you are ready for me to test things again on our end.

Reply:
The error I had before is because I used the same files to run the regtests and an external test, and the temporary compilation files were overwritten.

The regtests completed successfully including ww3_ts3 and ww3_tp2.15, and I guess that the small differences respect to the previous version of the branch is the extra code that has been included after merging to the last version of the development branch. I copy the differences here.

[matrixDiff.tar.gz](https://github.com/NOAA-EMC/WW3/files/5960247/matrixDiff.tar.gz)

In addition, I run a test using the Met Office AMM15 configuration and the results are consistent with the previous version of the branch. I have some small differences at some grid points, but again I attribute this to the changes after merging to the develop branch.

My conclusion is that the segmentation faults you have are something to do with your compiler or your machines, please check and I will be glad to give support if needed.

The multiple lines in the matrix.base for the ww3_tp2.15 are indeed there because the run_test script does not allow running multiple ww3_prnc in one test. This is the simplest solution I could think of, and it is similar to what I did for the new ww3_tp2.14 tests.

Reply:
@ukmo-juan-castillo so if I understand correctly, you are saying if I clean my compile, the code should then compile and run correctly for the cases I was getting segfaults?

Reply:
I do not know if that will work in your case, but it did in mine. If you keep having segfaults I would be interested in looking at the dumps.

Reply:
That did not help in my case.  It should run without any cleaning in between, when that's the cause of the issue is that w3_clean did not get updated after recent additions.    Are there any expected answer differences from this branch?  I don't remember the recent additions from the develop branch changing any answers so I'm curious why you are seeing answer differences after merging the develop branch in.   Since the bug is not reproducing on your end, I guess I'll have to turn on debug flags on my end and see if I can't figure out what's the cause.

Reply:
The differences I see before and after the merge look like changes in some regtests due to the new output field. I am afraid you will need to debug the code from your side, in the Met Office computer with the GNU compiler I do not have any problem. Tell me if I can help somehow.

Reply:
So, I'm not having much luck debugging so far.   

Right now, I'm getting this error: 
forrtl: error (72): floating overflow
Image              PC                Routine            Line        Source
ww3_shel           00000000011A620E  Unknown               Unknown  Unknown
libpthread-2.17.s  00002B8374D07630  Unknown               Unknown  Unknown
ww3_shel           0000000000928A48  w3updtmd_mp_w3uwn         460  w3updtmd.F90
libiomp5.so        00002B8375301A43  __kmp_invoke_micr     Unknown  Unknown
libiomp5.so        00002B83752C4CDA  Unknown               Unknown  Unknown
libiomp5.so        00002B83752C65B6  __kmp_fork_call       Unknown  Unknown
libiomp5.so        00002B8375284BB0  __kmpc_fork_call      Unknown  Unknown
ww3_shel           0000000000923AE6  w3updtmd_mp_w3uwn         452  w3updtmd.F90
ww3_shel           00000000008B71FE  w3wavemd_mp_w3wav         779  w3wavemd.F90
ww3_shel           0000000000466A8A  MAIN__                   1968  ww3_shel.F90

when running: 
./bin/run_test -b slurm -c hera -T -s ST2_PR2_UQ_MPI_OMPH -w work_ST2_PR2_UQ_MPI_OMPH -f -p mpirun -n 4 -t 6 ../model ww3_ts3

although I manually added the debug flags, so essentially the "hera" could be replaced with "intel_debug".   I get no errors when running this same test, with debugs on the develop branch.   The place where it says the error is, is not code that you changed or has been changed for a long time so I'm at a loss.   Any chance you could try to run this and the new test you added in debug mode on your system to see if you are also getting errors?  I'll continue to debug -- if anyone has suggestions I'd love to hear them.

Reply:
I can have a go at it on Monday. I will keep you informed on progress.

Reply:
I have tried different debug compiling options and everything works still. What are the problematic code lines? Might be you just have a memory problem, you could try increasing it.

Reply:
Well for the test:
./bin/run_test -b slurm -c hera -T -s ST2_PR2_UQ_MPI_OMPH -w work_ST2_PR2_UQ_MPI_OMPH -f -p mpirun -n 4 -t 6 ../model ww3_ts3
It can run in the develop branch w/debug flags turned on, but using this branch it fails.  The line it points to with error:
*** longjmp causes uninitialized stack frame ***: /scratch2/NCEPDEV/climate/Jessica.Meixner/PR_WW3/PR_fbcoupling/WW3_test/model/exe/ww3_shel terminated
....
forrtl: error (72): floating overflow
Image              PC                Routine            Line        Source
ww3_shel           00000000011A620E  Unknown               Unknown  Unknown
libpthread-2.17.s  00002B8374D07630  Unknown               Unknown  Unknown
ww3_shel           0000000000928A48  w3updtmd_mp_w3uwn         460  w3updtmd.F90
libiomp5.so        00002B8375301A43  __kmp_invoke_micr     Unknown  Unknown
libiomp5.so        00002B83752C4CDA  Unknown               Unknown  Unknown
libiomp5.so        00002B83752C65B6  __kmp_fork_call       Unknown  Unknown
libiomp5.so        00002B8375284BB0  __kmpc_fork_call      Unknown  Unknown
ww3_shel           0000000000923AE6  w3updtmd_mp_w3uwn         452  w3updtmd.F90
ww3_shel           00000000008B71FE  w3wavemd_mp_w3wav         779  w3wavemd.F90
ww3_shel           0000000000466A8A  MAIN__                   1968  ww3_shel.F90

which line 460 in the original is 
https://github.com/ukmo-waves/WW3/blob/fb_coupling_fields/model/ftn/w3updtmd.ftn#L584

I have been working to debug this test case since I can run it with the develop branch for a point of comparison.  I can try to increase the memory, but since every other regression test works and the develop branch works, that seems unlikely to be the issue to me.   I'll continue to debug and let you know what I find.

Reply:
@ukmo-juan-castillo I have traced it back to the fact that a call to W3FLDM in ww3_shel: https://github.com/ukmo-waves/WW3/blob/fb_coupling_fields/model/ftn/ww3_shel.ftn#L2348-L2350 when I run the develop branch w/debug flags and write out GA0, GD0, GAN, GDN after that call they are all 0, but in your branch for whatever reason they are very small or large values.  I'm trying to track down where/why this might be different between the two and will continue to keep you updated on my progress.